### DISCUSSION TITLE
[SFC] SD The great battle (Japan) overdumped on HW5 FW12.4

### DISCUSSION THREAD

**Discussion by Prominos** [sanni/cartreader#791] [Coding Corner]
Hello,

I just wanted to report that the game `SD The great battle` is overdumped on HW5 FW12.4. It's one of these games with a weird size so it doesn't surprise me.

The correct size is 786432 bytes or a 6M ROM.
The OSCR dumped/detected 1048576 bytes or a 8M ROM.

The checksum for this game is `38DC` it might need a custom config to set the correct number of memory banks.

```
OSCR HW5 V12.4

[+] Super Nintendo/SFC

[+] SNES/SFC cartridge


Searching database...
Checksum: 38DC
Header CRC32: 6565BD62
Found

Title: SD THE GREATBATTLE
Revision: 0
Type: LoROM SlowROM
ICs: ROM ONLY
ROM Size: 1 MB (32 banks)
Save Size: 0 KB
Checksum: 38DC
Press Button...

[+] Read ROM


Saving to SNES/ROM/SD THE GREATBATTLE/136/...
[*******************]
Checksum... 38DC -> OK
CRC32... AAE842D2 -> Not found
```
![20230516_202509](https://github.com/sanni/cartreader/assets/133436648/b88bdd00-ed5c-4ef1-9082-35fe2332be47)
![20230516_202745](https://github.com/sanni/cartreader/assets/133436648/80ed0e19-42e3-429e-abf0-fd5149633cad)

**Comment by PsyK0p4T** [comment]
Actually... I think OSCR is detecting it correctly, and Datomatic values could be bad.
That PCB is using a single chip, so the rom size should be a power of 2. 
If the used rom size is 6M, then the chip should be 8M for me.

Also, that reference "UPD23C8013CZ" is also used by other games, like "Super Ghouls'n'Ghosts" which is 8M.
https://datomatic.no-intro.org/index.php?page=show_record&s=49&n=2690

![image](https://github.com/sanni/cartreader/assets/87064902/1a5be7f4-1310-4974-85ab-0ea2435b0126)

**Comment by Prominos** [comment]
Interesting... They have an 8M ROM dump tagged as bad and overdumped, I wonder why that is? I'll check with them!

**Reply by Prominos** [reply]
You are absolutely right and I am now 99.9% sure this is an error on no-intro's end. I have explained it here if you have an account https://forum.no-intro.org/viewtopic.php?p=33671#p33671
